src/go/rpk/pkg/api/admin/api_cloud_storage_test.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type (
+	testRequestHandler func(w http.ResponseWriter, r *http.Request)
+	testFn             func(t *testing.T) testRequestHandler
+	requestTestCase    struct {
+		name             string
+		testFn           testFn
+		expectedResponse interface{}
+	}
+)
+
+func TestStartAutomatedRecovery(t *testing.T) {
+	successfullStartResponse := RecoveryStartResponse{
+		Code:    200,
+		Message: "Automated recovery started",
+	}
+
+	runTest := func(t *testing.T, test requestTestCase) {
+		baseURL := "http://non-existent-url"
+
+		if test.testFn != nil {
+			server := httptest.NewServer(http.HandlerFunc(test.testFn(t)))
+			defer server.Close()
+			baseURL = server.URL
+		}
+		client, err := NewAdminAPI([]string{baseURL}, BasicCredentials{}, nil)
+		assert.NoError(t, err)
+
+		response, err := client.StartAutomatedRecovery(context.Background(), ".*")
+
+		assert.NoError(t, err)
+		assert.Equal(t, test.expectedResponse, response)
+	}
+
+	tests := []requestTestCase{
+		{
+			name: "should call the correct endpoint",
+			testFn: func(t *testing.T) testRequestHandler {
+				return func(w http.ResponseWriter, r *http.Request) {
```

Comment:
This is better than whatever patten I used in the past, good pattern, I need to remember this.